File: Chat/Themes/Objective-c/ThemesViewController.m
//
//  ThemesViewController.m
//  Chat
//
//  Created by Qurban on 04.03.2019.
//  Copyright © 2019 Qurban. All rights reserved.
//

#import "ThemesViewController.h"

@interface ThemesViewController ()

@end

@implementation ThemesViewController

@synthesize model = _model;
@synthesize delegate = _delegate;

- (IBAction)theme1Action:(UIButton *)sender {
    self.view.backgroundColor = self.model.theme1;
    [self.delegate themesViewController:self didSelectTheme:self.model.theme1];
    [self.navigationController.navigationBar setBarTintColor:self.model.theme1];
}

- (IBAction)theme2Action:(UIButton *)sender {
    self.view.backgroundColor = self.model.theme2;
    [self.delegate themesViewController:self didSelectTheme:self.model.theme2];
    [self.navigationController.navigationBar setBarTintColor:self.model.theme2];
}

- (IBAction)theme3Action:(UIButton *)sender {
    self.view.backgroundColor = self.model.theme3;
    [self.delegate themesViewController:self didSelectTheme:self.model.theme3];
    [self.navigationController.navigationBar setBarTintColor:self.model.theme3];
}

- (IBAction)cancelAction:(UIBarButtonItem *)sender {
    [self dismissViewControllerAnimated:true completion:nil ];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    NSData *colorData = [NSUserDefaults.standardUserDefaults dataForKey:@"Theme"];
    UIColor *color = [NSKeyedUnarchiver unarchiveObjectWithData:colorData];
    if (color != nil) {
        self.view.backgroundColor = color;
    }
}

// get
- (Themes *)model {
    return _model;
}

// set
- (void)setModel:(Themes *)model {
    [model retain];
    [_model release];
    _model = model;
}

- (id<ThemesViewControllerDelegate>)delegate {
    return _delegate;
}

- (void)setDelegate:(id<ThemesViewControllerDelegate>)delegate {
    _delegate = delegate;
}


- (void)dealloc {
    [self.model release];
    [super dealloc];
}

@end

File: Chat/Themes/Swift/ThemesViewController.swift
//
//  ThemesViewController.swift
//  Chat
//
//  Created by Qurban on 06.03.2019.
//  Copyright © 2019 Qurban. All rights reserved.
//

import UIKit

class ThemesViewController: UIViewController {

    var closure: ((UIColor) -> Void)?

    var model: Themes!

    @IBAction func cancelAction(_ sender: UIBarButtonItem) {
        self.dismiss(animated: true, completion: nil)
    }

    @IBAction func theme1Action(_ sender: UIButton) {
        self.closure?(self.model.theme1)
        view.backgroundColor = self.model.theme1
        navigationController?.navigationBar.barTintColor = self.model.theme1
    }

    @IBAction func theme2Action(_ sender: UIButton) {
        self.closure?(self.model.theme2)
        view.backgroundColor = self.model.theme2
        navigationController?.navigationBar.barTintColor = self.model.theme2
    }

    @IBAction func theme3Action(_ sender: UIButton) {
        self.closure?(self.model.theme3)
        view.backgroundColor = self.model.theme3
        navigationController?.navigationBar.barTintColor = self.model.theme3
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        if let colorData = UserDefaults.standard.data(forKey: "Theme"),
            let color = NSKeyedUnarchiver.unarchiveObject(with: colorData) as? UIColor {
            view.backgroundColor = color
        }
    }

}
